\documentclass[]{article}
\usepackage{lmodern}
\usepackage{amssymb,amsmath}
\usepackage{ifxetex,ifluatex}
\usepackage{fixltx2e} % provides \textsubscript
\ifnum 0\ifxetex 1\fi\ifluatex 1\fi=0 % if pdftex
  \usepackage[T1]{fontenc}
  \usepackage[utf8]{inputenc}
\else % if luatex or xelatex
  \ifxetex
    \usepackage{mathspec}
  \else
    \usepackage{fontspec}
  \fi
  \defaultfontfeatures{Ligatures=TeX,Scale=MatchLowercase}
\fi
% use upquote if available, for straight quotes in verbatim environments
\IfFileExists{upquote.sty}{\usepackage{upquote}}{}
% use microtype if available
\IfFileExists{microtype.sty}{%
\usepackage{microtype}
\UseMicrotypeSet[protrusion]{basicmath} % disable protrusion for tt fonts
}{}


\usepackage{longtable,booktabs}
\usepackage{graphicx}
% grffile has become a legacy package: https://ctan.org/pkg/grffile
\IfFileExists{grffile.sty}{%
\usepackage{grffile}
}{}
\makeatletter
\def\maxwidth{\ifdim\Gin@nat@width>\linewidth\linewidth\else\Gin@nat@width\fi}
\def\maxheight{\ifdim\Gin@nat@height>\textheight\textheight\else\Gin@nat@height\fi}
\makeatother
% Scale images if necessary, so that they will not overflow the page
% margins by default, and it is still possible to overwrite the defaults
% using explicit options in \includegraphics[width, height, ...]{}
\setkeys{Gin}{width=\maxwidth,height=\maxheight,keepaspectratio}
\IfFileExists{parskip.sty}{%
\usepackage{parskip}
}{% else
\setlength{\parindent}{0pt}
\setlength{\parskip}{6pt plus 2pt minus 1pt}
}
\setlength{\emergencystretch}{3em}  % prevent overfull lines
\providecommand{\tightlist}{%
  \setlength{\itemsep}{0pt}\setlength{\parskip}{0pt}}
\setcounter{secnumdepth}{5}

%%% Use protect on footnotes to avoid problems with footnotes in titles
\let\rmarkdownfootnote\footnote%
\def\footnote{\protect\rmarkdownfootnote}

%%% Change title format to be more compact
\usepackage{titling}

% Create subtitle command for use in maketitle
\providecommand{\subtitle}[1]{
  \posttitle{
    \begin{center}\large#1\end{center}
    }
}

\setlength{\droptitle}{-2em}

\RequirePackage[]{/usr/local/lib/R/4.0/site-library/BiocStyle/resources/tex/Bioconductor}

\bioctitle[]{projectR Vignette}
    \pretitle{\vspace{\droptitle}\centering\huge}
  \posttitle{\par}
\author{Genevieve L. Stein-O'Brien}
    \preauthor{\centering\large\emph}
  \postauthor{\par}
      \predate{\centering\large\emph}
  \postdate{\par}
    \date{6 May 2021}

% code highlighting
\definecolor{fgcolor}{rgb}{0.251, 0.251, 0.251}
\newcommand{\hlnum}[1]{\textcolor[rgb]{0.816,0.125,0.439}{#1}}%
\newcommand{\hlstr}[1]{\textcolor[rgb]{0.251,0.627,0.251}{#1}}%
\newcommand{\hlcom}[1]{\textcolor[rgb]{0.502,0.502,0.502}{\textit{#1}}}%
\newcommand{\hlopt}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlstd}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwa}[1]{\textcolor[rgb]{0.125,0.125,0.941}{#1}}%
\newcommand{\hlkwb}[1]{\textcolor[rgb]{0,0,0}{#1}}%
\newcommand{\hlkwc}[1]{\textcolor[rgb]{0.251,0.251,0.251}{#1}}%
\newcommand{\hlkwd}[1]{\textcolor[rgb]{0.878,0.439,0.125}{#1}}%
\let\hlipl\hlkwb
%
\usepackage{fancyvrb}
\newcommand{\VerbBar}{|}
\newcommand{\VERB}{\Verb[commandchars=\\\{\}]}
\DefineVerbatimEnvironment{Highlighting}{Verbatim}{commandchars=\\\{\}}
%
\newenvironment{Shaded}{\begin{myshaded}}{\end{myshaded}}
% set background for result chunks
\let\oldverbatim\verbatim
\renewenvironment{verbatim}{\color{codecolor}\begin{myshaded}\begin{oldverbatim}}{\end{oldverbatim}\end{myshaded}}
%
\newcommand{\KeywordTok}[1]{\hlkwd{#1}}
\newcommand{\DataTypeTok}[1]{\hlkwc{#1}}
\newcommand{\DecValTok}[1]{\hlnum{#1}}
\newcommand{\BaseNTok}[1]{\hlnum{#1}}
\newcommand{\FloatTok}[1]{\hlnum{#1}}
\newcommand{\ConstantTok}[1]{\hlnum{#1}}
\newcommand{\CharTok}[1]{\hlstr{#1}}
\newcommand{\SpecialCharTok}[1]{\hlstr{#1}}
\newcommand{\StringTok}[1]{\hlstr{#1}}
\newcommand{\VerbatimStringTok}[1]{\hlstr{#1}}
\newcommand{\SpecialStringTok}[1]{\hlstr{#1}}
\newcommand{\ImportTok}[1]{{#1}}
\newcommand{\CommentTok}[1]{\hlcom{#1}}
\newcommand{\DocumentationTok}[1]{\hlcom{#1}}
\newcommand{\AnnotationTok}[1]{\hlcom{#1}}
\newcommand{\CommentVarTok}[1]{\hlcom{#1}}
\newcommand{\OtherTok}[1]{{#1}}
\newcommand{\FunctionTok}[1]{\hlstd{#1}}
\newcommand{\VariableTok}[1]{\hlstd{#1}}
\newcommand{\ControlFlowTok}[1]{\hlkwd{#1}}
\newcommand{\OperatorTok}[1]{\hlopt{#1}}
\newcommand{\BuiltInTok}[1]{{#1}}
\newcommand{\ExtensionTok}[1]{{#1}}
\newcommand{\PreprocessorTok}[1]{\textit{#1}}
\newcommand{\AttributeTok}[1]{{#1}}
\newcommand{\RegionMarkerTok}[1]{{#1}}
\newcommand{\InformationTok}[1]{\textcolor{messagecolor}{#1}}
\newcommand{\WarningTok}[1]{\textcolor{warningcolor}{#1}}
\newcommand{\AlertTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\ErrorTok}[1]{\textcolor{errorcolor}{#1}}
\newcommand{\NormalTok}[1]{\hlstd{#1}}
%
\AtBeginDocument{\bibliographystyle{/usr/local/lib/R/4.0/site-library/BiocStyle/resources/tex/unsrturl}}


\begin{document}
\maketitle


{
\setcounter{tocdepth}{2}
\tableofcontents
\newpage
}
\hypertarget{introduction}{%
\section{Introduction}\label{introduction}}

Technological advances continue to spur the exponential growth of biological data as illustrated by the rise of the omics---genomics, transcriptomics, epigenomics, proteomics, etc.---each with there own high throughput technologies. In order to leverage the full power of these resources, methods to integrate multiple data sets and data types must be developed. The reciprocal nature of the genomic, transcriptomic, epigenomic, and proteomic biology requires that the data provides a complementary view of cellular function and regulatory organization; however, the technical heterogeneity and massive size of high-throughput data even within a particular omic makes integrated analysis challenging. To address these challenges, we developed projectR, an R package for integrated analysis of high dimensional omic data. projectR uses the relationships defined within a given high dimensional data set, to interrogate related biological phenomena in an entirely new data set. By relying on relative comparisons within data type, projectR is able to circumvent many issues arising from technological variation. For a more extensive example of how the tools in the projectR package can be used for \emph{in silico} experiments, or additional information on the algorithm, see \href{https://www.sciencedirect.com/science/article/pii/S2405471219301462}{Stein-O'Brien, et al}.

\hypertarget{getting-started-with-projectr}{%
\section{Getting started with projectR}\label{getting-started-with-projectr}}

\hypertarget{installation-instructions}{%
\subsection{Installation Instructions}\label{installation-instructions}}

For automatic Bioconductor package installation, start R, and run:

\begin{verbatim}
BiocManager::install("projectR")
\end{verbatim}

\hypertarget{methods}{%
\subsection{Methods}\label{methods}}

Projection can roughly be defined as a mapping or transformation of points from one space to another often lower dimensional space. Mathematically, this can described as a function \(\varphi(x)=y : \Re^{D} \mapsto \Re^{d}\) s.t. \(d \leq D\) for \(x \in \Re^{D}, y \in \Re^{d}\) Barbakh, Wu, and Fyfe (2009) . The projectR package uses projection functions defined in a training dataset to interrogate related biological phenomena in an entirely new data set. These functions can be the product of any one of several methods common to ``omic'' analyses including regression, PCA, NMF, clustering. Individual sections focusing on one specific method are included in the vignette. However, the general design of the projectR function is the same regardless.

\hypertarget{the-base-projectr-function}{%
\subsection{The base projectR function}\label{the-base-projectr-function}}

The generic projectR function is executed as follows:

\begin{verbatim}
library(projectR)
projectR(data, loadings, dataNames=NULL, loadingsNames=NULL, NP = NULL, full = false)
\end{verbatim}

\hypertarget{input-arguments}{%
\subsubsection{Input Arguments}\label{input-arguments}}

The inputs that must be set each time are only the data and loadings, with all other inputs having default values. However, incongruities in the feature mapping between the data and loadings, i.e.~a different format for the rownames of each object, will throw errors or result in an empty mapping and should be checked before running. To overcoming mismatched feature names in the objects themselves, the /code\{dataNames\} and /code\{loadingNames\} arguments can be manually supplied by the user.

The arguments are as follows:

\begin{description}
\item[data]{a dataset to be projected into the pattern space}
\item[loadings]{a matrix of continous values with unique rownames to be projected}
\item[dataNames]{a vector containing unique name, i.e. gene names, for the rows of the target dataset to be used to match features with the loadings, if not provided by \texttt{rownames(data)}. Order of names in vector must match order of rows in data.}
\item[loadingsNames]{a vector containing unique names, i.e. gene names, for the rows of loadings to be used to match features with the data, if not provided by \texttt{rownames(loadings)}. Order of names in vector must match order of rows in loadings.}
\item[NP]{vector of integers indicating which columns of loadings object to use. The default of NP = NA will use entire matrix.}
\item[full]{logical indicating whether to return the full model solution. By default only the new pattern object is returned.}
\end{description}

The \texttt{loadings} argument in the generic projectR function is suitable for use with any genernal feature space, or set of feature spaces, whose rows annotation links them to the data to be projected. Ex: the coeffients associated with individual genes as the result of regression analysis or the amplituded values of individual genes as the result of non-negative matrix factorization (NMF).

\hypertarget{output}{%
\subsubsection{Output}\label{output}}

The basic output of the base projectR function, i.e.~\texttt{full=FALSE}, returns \texttt{projectionPatterns} representing relative weights for the samples from the new data in this previously defined feature space, or set of feature spaces. The full output of the base projectR function, i.e.~\texttt{full=TRUE}, returns \texttt{projectionFit}, a list containing \texttt{projectionPatterns} and \texttt{Projection}. The \texttt{Projection} object contains additional information from the proceedure used to obtain the \texttt{projectionPatterns}. For the the the base projectR function, \texttt{Projection} is the full lmFit model from the package \emph{\href{https://bioconductor.org/packages/3.12/limma}{limma}}.

\hypertarget{pca-projection}{%
\section{PCA projection}\label{pca-projection}}

Projection of principal components is achieved by matrix multiplication of a new data set by previously generated eigenvectors, or gene loadings. If the original data were standardized such that each gene is centered to zero average expression level, the principal components are normalized eigenvectors of the covariance matrix of the genes. Each PC is ordered according to how much of the variation present in the data they contain. Projection of the original samples into each PC will maximize the variance of the samples in the direction of that component and uncorrelated to previous components. Projection of new data places the new samples into the PCs defined by the original data. Because the components define an orthonormal basis set, they provide an isomorphism between a vector space, \(V\), and \(\Re^n\) which preserves inner products. If \(V\) is an inner product space over \(\Re\) with orthonormal basis \(B = v_1,...,v_n\) and \(v \epsilon V s.t [v]_B = (r_1,...,r_n)\), then finding the coordinate of \(v_i\) in \(v\) is precisely the inner product of \(v\) with \(v_i\), i.e.~\(r_i = \langle v,v_i \rangle\). This formulation is implemented for only those genes belonging to both the new data and the PC space. The \texttt{projectR} function has S4 method for class \texttt{prcomp}.

\hypertarget{obtaining-pcs-to-project.}{%
\subsection{Obtaining PCs to project.}\label{obtaining-pcs-to-project.}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# data to define PCs}
\FunctionTok{library}\NormalTok{(projectR)}
\FunctionTok{data}\NormalTok{(p.RNAseq6l3c3t)}

\CommentTok{\# do PCA on RNAseq6l3c3t expression data}
\NormalTok{pc.RNAseq6l3c3t}\OtherTok{\textless{}{-}}\FunctionTok{prcomp}\NormalTok{(}\FunctionTok{t}\NormalTok{(p.RNAseq6l3c3t))}
\NormalTok{pcVAR }\OtherTok{\textless{}{-}} \FunctionTok{round}\NormalTok{(((pc.RNAseq6l3c3t}\SpecialCharTok{$}\NormalTok{sdev)}\SpecialCharTok{\^{}}\DecValTok{2}\SpecialCharTok{/}\FunctionTok{sum}\NormalTok{(pc.RNAseq6l3c3t}\SpecialCharTok{$}\NormalTok{sdev}\SpecialCharTok{\^{}}\DecValTok{2}\NormalTok{))}\SpecialCharTok{*}\DecValTok{100}\NormalTok{,}\DecValTok{2}\NormalTok{)}
\NormalTok{dPCA }\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}\FunctionTok{cbind}\NormalTok{(pc.RNAseq6l3c3t}\SpecialCharTok{$}\NormalTok{x,pd.RNAseq6l3c3t))}

\CommentTok{\#plot pca}
\FunctionTok{library}\NormalTok{(ggplot2)}
\NormalTok{setCOL }\OtherTok{\textless{}{-}} \FunctionTok{scale\_colour\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"blue"}\NormalTok{,}\StringTok{"black"}\NormalTok{,}\StringTok{"red"}\NormalTok{), }\AttributeTok{name=}\StringTok{"Condition:"}\NormalTok{)}
\NormalTok{setFILL }\OtherTok{\textless{}{-}} \FunctionTok{scale\_fill\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"blue"}\NormalTok{,}\StringTok{"black"}\NormalTok{,}\StringTok{"red"}\NormalTok{),}\AttributeTok{guide =} \ConstantTok{FALSE}\NormalTok{)}
\NormalTok{setPCH }\OtherTok{\textless{}{-}} \FunctionTok{scale\_shape\_manual}\NormalTok{(}\AttributeTok{values=}\FunctionTok{c}\NormalTok{(}\DecValTok{23}\NormalTok{,}\DecValTok{22}\NormalTok{,}\DecValTok{25}\NormalTok{,}\DecValTok{25}\NormalTok{,}\DecValTok{21}\NormalTok{,}\DecValTok{24}\NormalTok{),}\AttributeTok{name=}\StringTok{"Cell Line:"}\NormalTok{)}

\NormalTok{pPCA }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(dPCA, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{PC1, }\AttributeTok{y=}\NormalTok{PC2, }\AttributeTok{colour=}\NormalTok{ID.cond, }\AttributeTok{shape=}\NormalTok{ID.line,}
        \AttributeTok{fill=}\NormalTok{ID.cond)) }\SpecialCharTok{+}
        \FunctionTok{geom\_point}\NormalTok{(}\FunctionTok{aes}\NormalTok{(}\AttributeTok{size=}\NormalTok{days),}\AttributeTok{alpha=}\NormalTok{.}\DecValTok{6}\NormalTok{)}\SpecialCharTok{+}
\NormalTok{        setCOL }\SpecialCharTok{+}\NormalTok{ setPCH  }\SpecialCharTok{+}\NormalTok{ setFILL }\SpecialCharTok{+}
        \FunctionTok{scale\_size\_area}\NormalTok{(}\AttributeTok{breaks =} \FunctionTok{c}\NormalTok{(}\DecValTok{2}\NormalTok{,}\DecValTok{4}\NormalTok{,}\DecValTok{6}\NormalTok{), }\AttributeTok{name=}\StringTok{"Day"}\NormalTok{) }\SpecialCharTok{+}
        \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position=}\FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{), }\AttributeTok{legend.justification=}\FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{),}
              \AttributeTok{legend.direction =} \StringTok{"horizontal"}\NormalTok{,}
              \AttributeTok{panel.background =} \FunctionTok{element\_rect}\NormalTok{(}\AttributeTok{fill =} \StringTok{"white"}\NormalTok{,}\AttributeTok{colour=}\ConstantTok{NA}\NormalTok{),}
              \AttributeTok{legend.background =} \FunctionTok{element\_rect}\NormalTok{(}\AttributeTok{fill =} \StringTok{"transparent"}\NormalTok{,}\AttributeTok{colour=}\ConstantTok{NA}\NormalTok{),}
              \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{vjust =} \DecValTok{0}\NormalTok{,}\AttributeTok{hjust=}\DecValTok{0}\NormalTok{,}\AttributeTok{face=}\StringTok{"bold"}\NormalTok{)) }\SpecialCharTok{+}
        \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"PCA of hPSC PolyA RNAseq"}\NormalTok{,}
            \AttributeTok{x=}\FunctionTok{paste}\NormalTok{(}\StringTok{"PC1 ("}\NormalTok{,pcVAR[}\DecValTok{1}\NormalTok{],}\StringTok{"\% of varience)"}\NormalTok{,}\AttributeTok{sep=}\StringTok{""}\NormalTok{),}
            \AttributeTok{y=}\FunctionTok{paste}\NormalTok{(}\StringTok{"PC2 ("}\NormalTok{,pcVAR[}\DecValTok{2}\NormalTok{],}\StringTok{"\% of varience)"}\NormalTok{,}\AttributeTok{sep=}\StringTok{""}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\hypertarget{projecting-prcomp-objects}{%
\subsection{Projecting prcomp objects}\label{projecting-prcomp-objects}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# data to project into PCs from RNAseq6l3c3t expression data}
\FunctionTok{data}\NormalTok{(p.ESepiGen4c1l)}

\FunctionTok{library}\NormalTok{(projectR)}
\NormalTok{PCA2ESepi }\OtherTok{\textless{}{-}} \FunctionTok{projectR}\NormalTok{(}\AttributeTok{data =}\NormalTok{ p.ESepiGen4c1l}\SpecialCharTok{$}\NormalTok{mRNA.Seq,}\AttributeTok{loadings=}\NormalTok{pc.RNAseq6l3c3t,}
\AttributeTok{full=}\ConstantTok{TRUE}\NormalTok{, }\AttributeTok{dataNames=}\NormalTok{map.ESepiGen4c1l[[}\StringTok{"GeneSymbols"}\NormalTok{]])}
\DocumentationTok{\#\# [1] "93 row names matched between data and loadings"}
\DocumentationTok{\#\# [1] "Updated dimension of data: 93 9"}

\NormalTok{pd.ESepiGen4c1l}\OtherTok{\textless{}{-}}\FunctionTok{data.frame}\NormalTok{(}\AttributeTok{Condition=}\FunctionTok{sapply}\NormalTok{(}\FunctionTok{colnames}\NormalTok{(p.ESepiGen4c1l}\SpecialCharTok{$}\NormalTok{mRNA.Seq),}
  \ControlFlowTok{function}\NormalTok{(x) }\FunctionTok{unlist}\NormalTok{(}\FunctionTok{strsplit}\NormalTok{(x,}\StringTok{\textquotesingle{}\_\textquotesingle{}}\NormalTok{))[}\DecValTok{1}\NormalTok{]),}\AttributeTok{stringsAsFactors=}\ConstantTok{FALSE}\NormalTok{)}
\NormalTok{pd.ESepiGen4c1l}\SpecialCharTok{$}\NormalTok{color}\OtherTok{\textless{}{-}}\FunctionTok{c}\NormalTok{(}\FunctionTok{rep}\NormalTok{(}\StringTok{"red"}\NormalTok{,}\DecValTok{2}\NormalTok{),}\FunctionTok{rep}\NormalTok{(}\StringTok{"green"}\NormalTok{,}\DecValTok{3}\NormalTok{),}\FunctionTok{rep}\NormalTok{(}\StringTok{"blue"}\NormalTok{,}\DecValTok{2}\NormalTok{),}\FunctionTok{rep}\NormalTok{(}\StringTok{"black"}\NormalTok{,}\DecValTok{2}\NormalTok{))}
\FunctionTok{names}\NormalTok{(pd.ESepiGen4c1l}\SpecialCharTok{$}\NormalTok{color)}\OtherTok{\textless{}{-}}\NormalTok{pd.ESepiGen4c1l}\SpecialCharTok{$}\NormalTok{Cond}

\NormalTok{dPCA2ESepi}\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}\FunctionTok{cbind}\NormalTok{(}\FunctionTok{t}\NormalTok{(PCA2ESepi[[}\DecValTok{1}\NormalTok{]]),pd.ESepiGen4c1l))}

\CommentTok{\#plot pca}
\FunctionTok{library}\NormalTok{(ggplot2)}
\NormalTok{setEpiCOL }\OtherTok{\textless{}{-}} \FunctionTok{scale\_colour\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"red"}\NormalTok{,}\StringTok{"green"}\NormalTok{,}\StringTok{"blue"}\NormalTok{,}\StringTok{"black"}\NormalTok{),}
  \AttributeTok{guide =} \FunctionTok{guide\_legend}\NormalTok{(}\AttributeTok{title=}\StringTok{"Lineage"}\NormalTok{))}

\NormalTok{pPC2ESepiGen4c1l }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(dPCA2ESepi, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{PC1, }\AttributeTok{y=}\NormalTok{PC2, }\AttributeTok{colour=}\NormalTok{Condition)) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{size=}\DecValTok{5}\NormalTok{) }\SpecialCharTok{+}\NormalTok{ setEpiCOL }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position=}\FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{), }\AttributeTok{legend.justification=}\FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{),}
  \AttributeTok{panel.background =} \FunctionTok{element\_rect}\NormalTok{(}\AttributeTok{fill =} \StringTok{"white"}\NormalTok{),}
  \AttributeTok{legend.direction =} \StringTok{"horizontal"}\NormalTok{,}
  \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{vjust =} \DecValTok{0}\NormalTok{,}\AttributeTok{hjust=}\DecValTok{0}\NormalTok{,}\AttributeTok{face=}\StringTok{"bold"}\NormalTok{)) }\SpecialCharTok{+}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"Encode RNAseq in target PC1 \& PC2"}\NormalTok{,}
  \AttributeTok{x=}\FunctionTok{paste}\NormalTok{(}\StringTok{"Projected PC1 ("}\NormalTok{,}\FunctionTok{round}\NormalTok{(PCA2ESepi[[}\DecValTok{2}\NormalTok{]][}\DecValTok{1}\NormalTok{],}\DecValTok{2}\NormalTok{),}\StringTok{"\% of varience)"}\NormalTok{,}\AttributeTok{sep=}\StringTok{""}\NormalTok{),}
  \AttributeTok{y=}\FunctionTok{paste}\NormalTok{(}\StringTok{"Projected PC2 ("}\NormalTok{,}\FunctionTok{round}\NormalTok{(PCA2ESepi[[}\DecValTok{2}\NormalTok{]][}\DecValTok{2}\NormalTok{],}\DecValTok{2}\NormalTok{),}\StringTok{"\% of varience)"}\NormalTok{,}\AttributeTok{sep=}\StringTok{""}\NormalTok{))}
\end{Highlighting}
\end{Shaded}

\begin{adjustwidth}{\fltoffset}{0mm}
\includegraphics[width=1\linewidth,]{projectR_files/figure-latex/unnamed-chunk-3-1} \end{adjustwidth}

\hypertarget{nmf-projection}{%
\section{NMF projection}\label{nmf-projection}}

NMF decomposes a data matrix of \(D\) with \(N\) genes as rows and \(M\) samples as columns, into two matrices, as \(D ~ AP\). The pattern matrix P has rows associated with BPs in samples and the amplitude matrix A has columns indicating the relative association of a given gene, where the total number of BPs (k) is an input parameter. CoGAPS and GWCoGAPS seek a pattern matrix (\({\bf{P}}\)) and the corresponding distribution matrix of weights (\({\bf{A}}\)) whose product forms a mock data matrix (\({\bf{M}}\)) that represents the gene-wise data \({\bf{D}}\) within noise limits (\(\boldsymbol{\varepsilon}\)). That is,
\begin{equation}
{\bf{D}} = {\bf{M}} + \boldsymbol{\varepsilon} = {\bf{A}}{\bf{P}} + \boldsymbol{\varepsilon}.
\label{eq:matrixDecomp}
\end{equation}
The number of rows in \({\bf{P}}\) (columns in \({\bf{A}}\)) defines the number of biological patterns (k) that CoGAPS/GWCoGAPS will infer from the number of nonorthogonal basis vectors required to span the data space. As in the Bayesian Decomposition algorithm Wang, Kossenkov, and Ochs (2006), the matrices \({\bf{A}}\) and \({\bf{P}}\) in CoGAPS are assumed to have the atomic prior described in Sibisi and Skilling (1997). In the CoGAPS/GWCoGAPS implementation, \(\alpha_{A}\) and \(\alpha_{P}\) are corresponding parameters for the expected number of atoms which map to each matrix element in \({\bf{A}}\) and \({\bf{P}}\), respectively. The corresponding matrices \({\bf{A}}\) and \({\bf{P}}\) are found by MCMC sampling.

Projection of CoGAPS/GWCoGAPS patterns is implemented by solving the factorization in \ref{eq:matrixDecomp} for the new data matrix where \({\bf{A}}\) is the fixed nonorthogonal basis vectors comprising the average of the posterior mean for the CoGAPS/GWCoGAPS simulations performed on the original data. The patterns \({\bf{P}}\) in the new data associated with this amplitude matrix is estimated using the least-squares fit to the new data implemented with the lmFit function in the \emph{\href{https://bioconductor.org/packages/3.12/limma}{limma}} package. The \texttt{projectR} function has S4 method for class \texttt{Linear Embedding Matrix, LME}.

\begin{verbatim}
library(projectR)
projectR(data, loadings,dataNames = NULL, loadingsNames = NULL,
     NP = NA, full = FALSE)
\end{verbatim}

\hypertarget{input-arguments-1}{%
\subsubsection{Input Arguments}\label{input-arguments-1}}

The inputs that must be set each time are only the data and patterns, with all other inputs having default values. However, inconguities between gene names--rownames of the loadings object and either rownames of the data object will throw errors and, subsequently, should be checked before running.

The arguments are as follows:

\begin{description}
\item[data]{a target dataset to be projected into the pattern space}
\item[loadings]{a CogapsResult object}
\item[dataNames]{rownames (eg. gene names) of the target dataset, if different from existing rownames of data}
\item[loadingsNames] loadingsNames rownames (eg. gene names) of the loadings to be matched with dataNames
\item[NP]{vector of integers indicating which columns of loadings object to use. The default of NP = NA will use entire matrix.}
\item[full]{logical indicating whether to return the full model solution. By default only the new pattern object is returned.}
\end{description}

\hypertarget{output-1}{%
\subsubsection{Output}\label{output-1}}

The basic output of the base projectR function, i.e.~\texttt{full=FALSE}, returns \texttt{projectionPatterns} representing relative weights for the samples from the new data in this previously defined feature space, or set of feature spaces. The full output of the base projectR function, i.e.~\texttt{full=TRUE}, returns \texttt{projectionFit}, a list containing \texttt{projectionPatterns} and \texttt{Projection}. The \texttt{Projection} object contains additional information from the procedure used to obtain the \texttt{projectionPatterns}. For the the the base projectR function, \texttt{Projection} is the full lmFit model from the package \emph{\href{https://bioconductor.org/packages/3.12/limma}{limma}}.

\hypertarget{obtaining-cogaps-patterns-to-project.}{%
\subsection{Obtaining CoGAPS patterns to project.}\label{obtaining-cogaps-patterns-to-project.}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# get data}
\FunctionTok{library}\NormalTok{(projectR)}
\NormalTok{AP }\OtherTok{\textless{}{-}} \FunctionTok{get}\NormalTok{(}\FunctionTok{data}\NormalTok{(}\StringTok{"AP.RNAseq6l3c3t"}\NormalTok{)) }\CommentTok{\#CoGAPS run data}
\NormalTok{AP }\OtherTok{\textless{}{-}}\NormalTok{ AP}\SpecialCharTok{$}\NormalTok{Amean}
\CommentTok{\# heatmap of gene weights for CoGAPs patterns}
\FunctionTok{library}\NormalTok{(gplots)}
\DocumentationTok{\#\# }
\DocumentationTok{\#\# Attaching package: \textquotesingle{}gplots\textquotesingle{}}
\DocumentationTok{\#\# The following object is masked from \textquotesingle{}package:stats\textquotesingle{}:}
\DocumentationTok{\#\# }
\DocumentationTok{\#\#     lowess}
\NormalTok{pNMF}\OtherTok{\textless{}{-}}\FunctionTok{heatmap.2}\NormalTok{(}\FunctionTok{as.matrix}\NormalTok{(AP),}\AttributeTok{col=}\NormalTok{bluered, }\AttributeTok{trace=}\StringTok{\textquotesingle{}none\textquotesingle{}}\NormalTok{,}
          \AttributeTok{distfun=}\ControlFlowTok{function}\NormalTok{(c) }\FunctionTok{as.dist}\NormalTok{(}\DecValTok{1}\SpecialCharTok{{-}}\FunctionTok{cor}\NormalTok{(}\FunctionTok{t}\NormalTok{(c))) ,}
          \AttributeTok{cexCol=}\DecValTok{1}\NormalTok{,}\AttributeTok{cexRow=}\NormalTok{.}\DecValTok{5}\NormalTok{,}\AttributeTok{scale =} \StringTok{"row"}\NormalTok{,}
          \AttributeTok{hclustfun=}\ControlFlowTok{function}\NormalTok{(x) }\FunctionTok{hclust}\NormalTok{(x, }\AttributeTok{method=}\StringTok{"average"}\NormalTok{)}
\NormalTok{      )}
\end{Highlighting}
\end{Shaded}

\begin{adjustwidth}{\fltoffset}{0mm}
\includegraphics[width=1\linewidth,]{projectR_files/figure-latex/unnamed-chunk-4-1} \end{adjustwidth}

\hypertarget{projecting-cogaps-objects}{%
\subsection{Projecting CoGAPS objects}\label{projecting-cogaps-objects}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# data to project into PCs from RNAseq6l3c3t expression data}
\FunctionTok{library}\NormalTok{(projectR)}
\FunctionTok{data}\NormalTok{(}\StringTok{\textquotesingle{}p.ESepiGen4c1l4\textquotesingle{}}\NormalTok{)}
\DocumentationTok{\#\# Warning in data("p.ESepiGen4c1l4"): data set \textquotesingle{}p.ESepiGen4c1l4\textquotesingle{} not found}
\FunctionTok{data}\NormalTok{(}\StringTok{\textquotesingle{}p.RNAseq6l3c3t\textquotesingle{}}\NormalTok{)}

\NormalTok{NMF2ESepi }\OtherTok{\textless{}{-}} \FunctionTok{projectR}\NormalTok{(p.ESepiGen4c1l}\SpecialCharTok{$}\NormalTok{mRNA.Seq,}\AttributeTok{loadings=}\NormalTok{AP,}\AttributeTok{full=}\ConstantTok{TRUE}\NormalTok{,}
    \AttributeTok{dataNames=}\NormalTok{map.ESepiGen4c1l[[}\StringTok{"GeneSymbols"}\NormalTok{]])}
\DocumentationTok{\#\# [1] "93 row names matched between data and loadings"}
\DocumentationTok{\#\# [1] "Updated dimension of data: 93 9"}

\NormalTok{dNMF2ESepi}\OtherTok{\textless{}{-}} \FunctionTok{data.frame}\NormalTok{(}\FunctionTok{cbind}\NormalTok{(}\FunctionTok{t}\NormalTok{(NMF2ESepi),pd.ESepiGen4c1l))}

\CommentTok{\#plot pca}
\FunctionTok{library}\NormalTok{(ggplot2)}
\NormalTok{setEpiCOL }\OtherTok{\textless{}{-}} \FunctionTok{scale\_colour\_manual}\NormalTok{(}\AttributeTok{values =} \FunctionTok{c}\NormalTok{(}\StringTok{"red"}\NormalTok{,}\StringTok{"green"}\NormalTok{,}\StringTok{"blue"}\NormalTok{,}\StringTok{"black"}\NormalTok{),}
\AttributeTok{guide =} \FunctionTok{guide\_legend}\NormalTok{(}\AttributeTok{title=}\StringTok{"Lineage"}\NormalTok{))}

\NormalTok{pNMF2ESepiGen4c1l }\OtherTok{\textless{}{-}} \FunctionTok{ggplot}\NormalTok{(dNMF2ESepi, }\FunctionTok{aes}\NormalTok{(}\AttributeTok{x=}\NormalTok{X1, }\AttributeTok{y=}\NormalTok{X2, }\AttributeTok{colour=}\NormalTok{Condition)) }\SpecialCharTok{+}
  \FunctionTok{geom\_point}\NormalTok{(}\AttributeTok{size=}\DecValTok{5}\NormalTok{) }\SpecialCharTok{+}\NormalTok{ setEpiCOL }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}\AttributeTok{legend.position=}\FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{), }\AttributeTok{legend.justification=}\FunctionTok{c}\NormalTok{(}\DecValTok{0}\NormalTok{,}\DecValTok{0}\NormalTok{),}
  \AttributeTok{panel.background =} \FunctionTok{element\_rect}\NormalTok{(}\AttributeTok{fill =} \StringTok{"white"}\NormalTok{),}
  \AttributeTok{legend.direction =} \StringTok{"horizontal"}\NormalTok{,}
  \AttributeTok{plot.title =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{vjust =} \DecValTok{0}\NormalTok{,}\AttributeTok{hjust=}\DecValTok{0}\NormalTok{,}\AttributeTok{face=}\StringTok{"bold"}\NormalTok{))}
  \FunctionTok{labs}\NormalTok{(}\AttributeTok{title =} \StringTok{"Encode RNAseq in target PC1 \& PC2"}\NormalTok{,}
  \AttributeTok{x=}\FunctionTok{paste}\NormalTok{(}\StringTok{"Projected PC1 ("}\NormalTok{,}\FunctionTok{round}\NormalTok{(PCA2ESepi[[}\DecValTok{2}\NormalTok{]][}\DecValTok{1}\NormalTok{],}\DecValTok{2}\NormalTok{),}\StringTok{"\% of varience)"}\NormalTok{,}\AttributeTok{sep=}\StringTok{""}\NormalTok{),}
  \AttributeTok{y=}\FunctionTok{paste}\NormalTok{(}\StringTok{"Projected PC2 ("}\NormalTok{,}\FunctionTok{round}\NormalTok{(PCA2ESepi[[}\DecValTok{2}\NormalTok{]][}\DecValTok{2}\NormalTok{],}\DecValTok{2}\NormalTok{),}\StringTok{"\% of varience)"}\NormalTok{,}\AttributeTok{sep=}\StringTok{""}\NormalTok{))}
\DocumentationTok{\#\# $x}
\DocumentationTok{\#\# [1] "Projected PC1 (18.33\% of varience)"}
\DocumentationTok{\#\# }
\DocumentationTok{\#\# $y}
\DocumentationTok{\#\# [1] "Projected PC2 (17.13\% of varience)"}
\DocumentationTok{\#\# }
\DocumentationTok{\#\# $title}
\DocumentationTok{\#\# [1] "Encode RNAseq in target PC1 \& PC2"}
\DocumentationTok{\#\# }
\DocumentationTok{\#\# attr(,"class")}
\DocumentationTok{\#\# [1] "labels"}
\end{Highlighting}
\end{Shaded}

\hypertarget{clustering-projection}{%
\section{Clustering projection}\label{clustering-projection}}

As canonical projection is not defined for clustering objects, the projectR package offers two transfer learning inspired methods to achieve the ``projection'' of clustering objects. These methods are defined by the function used to quantify and transfer the relationships which define each cluster in the original data set to the new dataset. Briefly, \texttt{cluster2pattern} uses the corelation of each genes expression to the mean of each cluster to define continuous weights. These weights are output as a \texttt{pclust} object which can serve as input to \texttt{projectR}. Alternatively, the \texttt{intersectoR} function can be used to test for significant overlap between two clustering objects. Both \texttt{cluster2pattern} and \texttt{intersectoR} methods are coded for a generic list structure with additional S4 class methods for kmeans and hclust objects. Further details and examples are provided in the followin respecitive sections.

\hypertarget{cluster2pattern}{%
\subsection{cluster2pattern}\label{cluster2pattern}}

\texttt{cluster2pattern} uses the corelation of each genes expression to the mean of each cluster to define continuous weights.

\begin{verbatim}
library(projectR)
data(p.RNAseq6l3c3t)


nP<-5
kClust<-kmeans(p.RNAseq6l3c3t,centers=nP)
kpattern<-cluster2pattern(clusters = kClust, NP = nP, data = p.RNAseq6l3c3t)
kpattern

cluster2pattern(clusters = NA, NP = NA, data = NA)
\end{verbatim}

\hypertarget{input-arguments-2}{%
\subsubsection{Input Arguments}\label{input-arguments-2}}

The inputs that must be set each time are the clusters and data.

The arguments are as follows:

\begin{description}
\item[clusters]{a clustering object}
\item[NP]{either the number of clusters desired or the subset of clusters to use}
\item[data]{data used to make clusters object}
\end{description}

\hypertarget{output-2}{%
\subsubsection{Output}\label{output-2}}

The output of the \texttt{cluster2pattern} function is a \texttt{pclust} class object; specifically, a matrix of genes (rows) by clusters (columns). A gene's value outside of its assigned cluster is zero. For the cluster containing a given gene, the gene's value is the correlation of the gene's expression to the mean of that cluster.

\hypertarget{intersector}{%
\subsection{intersectoR}\label{intersector}}

\texttt{intersectoR} function can be used to test for significant overlap between two clustering objects. The base function finds and tests the intersecting values of two sets of lists, presumably the genes associated with patterns in two different datasets. S4 class methods for \texttt{hclust} and \texttt{kmeans} objects are also available.

\begin{verbatim}
library(projectR)
intersectoR(pSet1 = NA, pSet2 = NA, pval = 0.05, full = FALSE, k = NULL)
\end{verbatim}

\hypertarget{input-arguments-3}{%
\subsubsection{Input Arguments}\label{input-arguments-3}}

The inputs that must be set each time are the clusters and data.

The arguments are as follows:

\begin{description}
\item[pSet1]{a list for a set of patterns where each entry is a set of genes associated with a single pattern}
\item[pSet2]{a list for a second set of patterns where each entry is a set of genes associated with a single pattern}
\item[pval]{the maximum p-value considered significant}
\item[full]{logical indicating whether to return full data frame of signigicantly overlapping sets. Default is false will return summary matrix.}
\item[k]{numeric giving cut height for hclust objects, if vector arguments will be applied to pSet1 and pSet2 in that order}
\end{description}

\hypertarget{output-3}{%
\subsubsection{Output}\label{output-3}}

The output of the \texttt{intersectoR} function is a summary matrix showing the sets with statistically significant overlap under the specified \(p\)-value threshold based on a hypergeometric test. If \texttt{full==TRUE} the full data frame of signigicantly overlapping sets will also be returned.

\hypertarget{correlation-based-projection}{%
\section{Correlation based projection}\label{correlation-based-projection}}

Correlation based projection requires a matrix of gene-wise correlation values to serve as the Pattern input to the \texttt{projectR} function. This matrix can be user-generated or the result of the \texttt{correlateR} function included in the projectR package. User-generated matrixes with each row corresponding to an individual gene can be input to the generic \texttt{projectR} function. The \texttt{correlateR} function allows users to create a weight matrix for projection with values quantifying the within dataset correlation of each genes expression to the expression pattern of a particular gene or set of genes as follows.

\hypertarget{correlater}{%
\subsection{correlateR}\label{correlater}}

\begin{verbatim}
library(projectR)
correlateR(genes = NA, dat = NA, threshtype = "R", threshold = 0.7, absR = FALSE, ...)
\end{verbatim}

\hypertarget{input-arguments-4}{%
\subsubsection{Input Arguments}\label{input-arguments-4}}

The inputs that must be set each time are only the genes and data, with all other inputs having default values.

The arguments are as follows:

\begin{description}
\item[genes]{gene or character vector of genes for reference expression pattern dat}
\item[data]{matrix or data frame with genes to be used for to calculate correlation}
\item[threshtype]{Default "R" indicates thresholding by R value or equivalent. Alternatively, "N" indicates a numerical cut off.}
\item[threshold]{numeric indicating value at which to make threshold}
\item[absR]{logical indicating where to include both positive and negatively correlated genes}
\item[...]{addtion imputes to the cor function}
\end{description}

\hypertarget{output-4}{%
\subsubsection{Output}\label{output-4}}

The output of the \texttt{correlateR} function is a \texttt{correlateR} class object. Specifically, a matrix of correlation values for those genes whose expression pattern pattern in the dataset is correlated (and anti-correlated if absR=TRUE) above the value given in as the threshold arguement. As this information may be useful in its own right, it is recommended that users inspect the \texttt{correlateR} object before using it as input to the \texttt{projectR} function.

\hypertarget{obtaining-and-visualizing-objects.}{%
\subsection{\texorpdfstring{Obtaining and visualizing \texttt{correlateR} objects.}{Obtaining and visualizing  objects.}}\label{obtaining-and-visualizing-objects.}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# data to}
\FunctionTok{library}\NormalTok{(projectR)}
\FunctionTok{data}\NormalTok{(}\StringTok{"p.RNAseq6l3c3t"}\NormalTok{)}

\CommentTok{\# get genes correlated to T}
\NormalTok{cor2T}\OtherTok{\textless{}{-}}\FunctionTok{correlateR}\NormalTok{(}\AttributeTok{genes=}\StringTok{"T"}\NormalTok{, }\AttributeTok{dat=}\NormalTok{p.RNAseq6l3c3t, }\AttributeTok{threshtype=}\StringTok{"N"}\NormalTok{, }\AttributeTok{threshold=}\DecValTok{10}\NormalTok{, }\AttributeTok{absR=}\ConstantTok{TRUE}\NormalTok{)}
\NormalTok{cor2T }\OtherTok{\textless{}{-}}\NormalTok{ cor2T}\SpecialCharTok{@}\NormalTok{corM}
\DocumentationTok{\#\#\# heatmap of genes more correlated to T}
\NormalTok{indx}\OtherTok{\textless{}{-}}\FunctionTok{unlist}\NormalTok{(}\FunctionTok{sapply}\NormalTok{(cor2T,rownames))}
\NormalTok{indx }\OtherTok{\textless{}{-}} \FunctionTok{as.vector}\NormalTok{(indx)}
\FunctionTok{colnames}\NormalTok{(p.RNAseq6l3c3t)}\OtherTok{\textless{}{-}}\NormalTok{pd.RNAseq6l3c3t}\SpecialCharTok{$}\NormalTok{sampleX}
\FunctionTok{library}\NormalTok{(reshape2)}
\NormalTok{pm.RNAseq6l3c3t}\OtherTok{\textless{}{-}}\FunctionTok{melt}\NormalTok{(}\FunctionTok{cbind}\NormalTok{(p.RNAseq6l3c3t[indx,],indx))}
\DocumentationTok{\#\# Using indx as id variables}

\FunctionTok{library}\NormalTok{(gplots)}
\FunctionTok{library}\NormalTok{(ggplot2)}
\FunctionTok{library}\NormalTok{(viridis)}
\DocumentationTok{\#\# Loading required package: viridisLite}
\NormalTok{pCorT}\OtherTok{\textless{}{-}}\FunctionTok{ggplot}\NormalTok{(pm.RNAseq6l3c3t, }\FunctionTok{aes}\NormalTok{(variable, indx, }\AttributeTok{fill =}\NormalTok{ value)) }\SpecialCharTok{+}
  \FunctionTok{geom\_tile}\NormalTok{(}\AttributeTok{colour=}\StringTok{"gray20"}\NormalTok{, }\AttributeTok{size=}\FloatTok{1.5}\NormalTok{, }\AttributeTok{stat=}\StringTok{"identity"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_fill\_viridis}\NormalTok{(}\AttributeTok{option=}\StringTok{"B"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{xlab}\NormalTok{(}\StringTok{""}\NormalTok{) }\SpecialCharTok{+}  \FunctionTok{ylab}\NormalTok{(}\StringTok{""}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{scale\_y\_discrete}\NormalTok{(}\AttributeTok{limits=}\NormalTok{indx) }\SpecialCharTok{+}
  \FunctionTok{ggtitle}\NormalTok{(}\StringTok{"Ten genes most highly pos \& neg correlated with T"}\NormalTok{) }\SpecialCharTok{+}
  \FunctionTok{theme}\NormalTok{(}
    \AttributeTok{panel.background =} \FunctionTok{element\_rect}\NormalTok{(}\AttributeTok{fill=}\StringTok{"gray20"}\NormalTok{),}
    \AttributeTok{panel.border =} \FunctionTok{element\_rect}\NormalTok{(}\AttributeTok{fill=}\ConstantTok{NA}\NormalTok{,}\AttributeTok{color=}\StringTok{"gray20"}\NormalTok{, }\AttributeTok{size=}\FloatTok{0.5}\NormalTok{, }\AttributeTok{linetype=}\StringTok{"solid"}\NormalTok{),}
    \AttributeTok{panel.grid.major =} \FunctionTok{element\_blank}\NormalTok{(),}
    \AttributeTok{panel.grid.minor =} \FunctionTok{element\_blank}\NormalTok{(),}
    \AttributeTok{axis.line =} \FunctionTok{element\_blank}\NormalTok{(),}
    \AttributeTok{axis.ticks =} \FunctionTok{element\_blank}\NormalTok{(),}
    \AttributeTok{axis.text =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{size=}\FunctionTok{rel}\NormalTok{(}\DecValTok{1}\NormalTok{),}\AttributeTok{hjust=}\DecValTok{1}\NormalTok{),}
    \AttributeTok{axis.text.x =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{angle =} \DecValTok{90}\NormalTok{,}\AttributeTok{vjust=}\NormalTok{.}\DecValTok{5}\NormalTok{),}
    \AttributeTok{legend.text =} \FunctionTok{element\_text}\NormalTok{(}\AttributeTok{color=}\StringTok{"white"}\NormalTok{, }\AttributeTok{size=}\FunctionTok{rel}\NormalTok{(}\DecValTok{1}\NormalTok{)),}
    \AttributeTok{legend.background =} \FunctionTok{element\_rect}\NormalTok{(}\AttributeTok{fill=}\StringTok{"gray20"}\NormalTok{),}
    \AttributeTok{legend.position =} \StringTok{"bottom"}\NormalTok{,}
    \AttributeTok{legend.title=}\FunctionTok{element\_blank}\NormalTok{()}
\NormalTok{)}
\end{Highlighting}
\end{Shaded}

\begin{adjustwidth}{\fltoffset}{0mm}
\includegraphics[width=1\linewidth,]{projectR_files/figure-latex/unnamed-chunk-6-1} \end{adjustwidth}

\hypertarget{projecting-correlater-objects.}{%
\subsection{Projecting correlateR objects.}\label{projecting-correlater-objects.}}

\begin{Shaded}
\begin{Highlighting}[]
\CommentTok{\# data to project into from RNAseq6l3c3t expression data}
\FunctionTok{data}\NormalTok{(p.ESepiGen4c1l)}

\FunctionTok{library}\NormalTok{(projectR)}
\NormalTok{cor2ESepi }\OtherTok{\textless{}{-}} \FunctionTok{projectR}\NormalTok{(p.ESepiGen4c1l}\SpecialCharTok{$}\NormalTok{mRNA.Seq,}\AttributeTok{loadings=}\NormalTok{cor2T[[}\DecValTok{1}\NormalTok{]],}\AttributeTok{full=}\ConstantTok{FALSE}\NormalTok{,}
    \AttributeTok{dataNames=}\NormalTok{map.ESepiGen4c1l}\SpecialCharTok{$}\NormalTok{GeneSymbols)}
\DocumentationTok{\#\# [1] "9 row names matched between data and loadings"}
\DocumentationTok{\#\# [1] "Updated dimension of data: 9 9"}
\end{Highlighting}
\end{Shaded}

\hypertarget{refs}{}
\begin{CSLReferences}{1}{0}
\leavevmode\vadjust pre{\hypertarget{ref-Barbakh:2009bw}{}}%
Barbakh, Wesam Ashour, Ying Wu, and Colin Fyfe. 2009. {``{Review of Linear Projection Methods}.''} In \emph{Non-Standard Parameter Adaptation for Exploratory Data Analysis}, 29--48. Berlin, Heidelberg: Springer Berlin Heidelberg.

\leavevmode\vadjust pre{\hypertarget{ref-Sibisi1997}{}}%
Sibisi, Sibusiso, and John Skilling. 1997. {``Prior Distributions on Measure Space.''} \emph{Journal of the Royal Statistical Society: Series B (Statistical Methodology)} 59 (1): 217--35. \url{https://doi.org/10.1111/1467-9868.00065}.

\leavevmode\vadjust pre{\hypertarget{ref-Ochs2006}{}}%
Wang, Guoli, Andrew V. Kossenkov, and Michael F. Ochs. 2006. {``LS-NMF: A Modified Non-Negative Matrix Factorization Algorithm Utilizing Uncertainty Estimates.''} \emph{BMC Bioinformatics} 7 (1): 175. \url{https://doi.org/10.1186/1471-2105-7-175}.

\end{CSLReferences}


\end{document}
